File: src/main/java/com/baidu/algorithm/dp/_122_Best_Time_to_Buy_and_Sell_Stock_II.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if prices is None or len(prices) < 2:
            return 0

        max_profit = 0
        for i in range(1, len(prices)):
            max_profit += (prices[i] - prices[i - 1]) if  (prices[i] - prices[i - 1]) > 0 else 0

        return max_profit

File: src/main/java/com/baidu/algorithm/dp/_122_Best_Time_to_Buy_and_Sell_Stock_II.java
/*
 * Copyright (C) 2018 Baidu, Inc. All Rights Reserved.
 */
package com.baidu.algorithm.dp;

/**
 * _122_Best_Time_to_Buy_and_Sell_Stock_II
 *
 * @author xuhaoran01
 */
public class _122_Best_Time_to_Buy_and_Sell_Stock_II {

    public int maxProfit(int[] prices) {
        if(prices == null || prices.length == 0) {
            return 0;
        }

        int res = 0;
        for(int i = 1; i < prices.length; i++) {
            res += Math.max(0, prices[i] - prices[i - 1]);
        }

        return res;
    }
}
